File: Frontend/src/models/Inquiry.ts
import { ChatMessage } from "./ChatMessage";
import { InquiryImage } from "./InquiryImage";
import { InquiryStatus } from "./InquiryStatus";
import { User } from "./User";

export interface Inquiry {
	id: number;
	title: string;
	description: string;
	generalPractitioner: string;
	summary?: string;
	receivedDate?: string;
	updatedDate?: string;
	patientPersonNumber?: number;
	patient?: User;
	nursePersonNumber?: number;
	nurse?: User;
	inquiryStatusId: number;
	inquiryStatus?: InquiryStatus;
	inquiryImages?: InquiryImage[];
	chatMessages?: ChatMessage[];
}

export const INQUIRY_CONSTANTS = {
	INITIAL_INQUIRY: {
		id: 0,
		title: "string",
		description: "string",
		generalPractitioner: "string",
		inquiryStatusId: 0
	},

	ANALYTICS_PERIODS: {
		1: { text: "This year" },
		2: { text: "Last year" }
	},
	PAGINATION_OPTIONS: [10, 20, 30],
	ROWS_PER_PAGE: 10,
	TITLE_MINIMUM_VALID_LENGTH: 15,
	TITLE_MAXIMUM_VALID_LENGTH: 100,
	DESCRIPTION_MINIMUM_VALID_LENGTH: 25,
	GENERAL_PRACTITIONER_MINIMUM_VALID_LENGTH: 10,
	GENERAL_PRACTITIONER_MAXIMUM_VALID_LENGTH: 50
};

File: Backend/Src/Models/Inquiry.cs
using EmergencyDepartment.Constants;
using System.ComponentModel.DataAnnotations;

namespace EmergencyDepartment.Models;

public partial class Inquiry
{
	public int Id { get; set; }

	[MinLength(15), MaxLength(100)]
	public required string Title { get; set; }

	[MinLength(25)]
	public required string Description { get; set; }

	[MinLength(10), MaxLength(50)]
	public required string GeneralPractitioner { get; set; }

	public string? Summary { get; set; }

	public DateTime ReceivedDate { get; set; }

	public DateTime? UpdatedDate { get; set; }

	[RegularExpression(@"^\d{11}$", ErrorMessage = UserConstants.PersonNumberInvalidErrorMessage)]
	public long? PatientPersonNumber { get; set; }

	public virtual User? Patient { get; set; }

	[RegularExpression(@"^\d{11}$", ErrorMessage = UserConstants.PersonNumberInvalidErrorMessage)]
	public long? NursePersonNumber { get; set; }

	public virtual User? Nurse { get; set; }

	public int InquiryStatusId { get; set; }

	public virtual InquiryStatus? InquiryStatus { get; set; }

	public virtual ICollection<InquiryImage> InquiryImages { get; set; } = [];

	public virtual ICollection<ChatMessage> ChatMessages { get; set; } = [];
}